src/Tizen.Wearable.CircularUI.Forms/GoogleMapView.cs

```diff
@@ -0,0 +1,206 @@
+            }
+
+            _pins.Clear();
+            CreatePinItems();
+        }
+
+        void OnItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            switch (e.Action)
+            {
+                case NotifyCollectionChangedAction.Add:
+                    if (e.NewStartingIndex == -1)
+                        goto case NotifyCollectionChangedAction.Reset;
+                    foreach (object item in e.NewItems)
+                        CreatePin(item);
+                    break;
+                case NotifyCollectionChangedAction.Move:
+                    if (e.OldStartingIndex == -1 || e.NewStartingIndex == -1)
+                        goto case NotifyCollectionChangedAction.Reset;
+                    // Not tracking order
+                    break;
+                case NotifyCollectionChangedAction.Remove:
+                    if (e.OldStartingIndex == -1)
+                        goto case NotifyCollectionChangedAction.Reset;
+                    foreach (object item in e.OldItems)
+                        RemovePin(item);
+                    break;
+                case NotifyCollectionChangedAction.Replace:
+                    if (e.OldStartingIndex == -1)
+                        goto case NotifyCollectionChangedAction.Reset;
+                    foreach (object item in e.OldItems)
+                        RemovePin(item);
+                    foreach (object item in e.NewItems)
+                        CreatePin(item);
+                    break;
+                case NotifyCollectionChangedAction.Reset:
+                    _pins.Clear();
+                    break;
+            }
+        }
+
+        void CreatePinItems()
+        {
+            if (ItemsSource == null || ItemTemplate == null)
+            {
+                return;
+            }
+
+            foreach (object item in ItemsSource)
+            {
+                CreatePin(item);
+            }
+        }
+
+        void CreatePin(object newItem)
+        {
+            if (ItemTemplate == null)
+            {
+                return;
+            }
+
+            var pin = (Pin)ItemTemplate.CreateContent();
+            pin.BindingContext = newItem;
+            _pins.Add(pin);
+        }
+
+        void RemovePin(object itemToRemove)
+        {
+            Pin pinToRemove = _pins.FirstOrDefault(pin => pin.BindingContext?.Equals(itemToRemove) == true);
+            if (pinToRemove != null)
+            {
+                _pins.Remove(pinToRemove);
+            }
+        }
+
+        /// <summary>
+        /// Set GoogleMapOption value to MapView.
+        /// </summary>
+        public void SetMapOption(GoogleMapOption value)
+        {
+            MapOption = value;
+            LoadMapRequested?.Invoke(this, EventArgs.Empty);
```

Comment:
Why don't use `MapOptionEventArgs` ?

Reply:
For convenience of implementation, only forward the event. The actual GoogleMapOption  value is passed to MapOption.